File: frontend/app/types/modules.ts
// ============================================================================
// MODULE SYSTEM TYPES
// For production handoff to DCO/automation platforms
// (Flashtalking, Innovid, Clinch, Celtra, Storyteq, etc.)
// ============================================================================

// ============================================================================
// MODULE TAXONOMY
// ============================================================================

export type ModuleType = 
  | 'hook'          // Opening attention-grabber (first 3 seconds)
  | 'value_prop'    // Core value proposition
  | 'proof_point'   // Social proof, testimonials, data points
  | 'product'       // Product shot, demo, feature highlight
  | 'offer'         // Promotional offer, pricing
  | 'cta'           // Call to action (button, end card)
  | 'background'    // Visual background, texture, environment
  | 'logo'          // Brand logo treatment
  | 'legal'         // Disclaimers, terms, fine print
  | 'audio'         // Voiceover, music, SFX
  | 'end_card'      // Closing frame, brand lockup
  | 'transition';   // Motion between modules

export const MODULE_TYPE_CONFIG: Record<ModuleType, {
  label: string;
  description: string;
  color: string;
  icon: string;
  typical_duration?: string;
  common_variations: string[];
}> = {
  hook: {
    label: 'Hook',
    description: 'Opening attention-grabber in first 3 seconds',
    color: 'purple',
    icon: '🎯',
    typical_duration: '2-3s',
    common_variations: ['Emotional', 'Question', 'Statistic', 'Problem', 'Bold claim'],
  },
  value_prop: {
    label: 'Value Proposition',
    description: 'Core benefit or differentiator',
    color: 'blue',
    icon: '💎',
    common_variations: ['Functional', 'Emotional', 'Social', 'Financial'],
  },
  proof_point: {
    label: 'Proof Point',
    description: 'Evidence, testimonials, or social proof',
    color: 'green',
    icon: '✓',
    common_variations: ['Testimonial', 'Statistic', 'Award', 'Review', 'Case study'],
  },
  product: {
    label: 'Product',
    description: 'Product visualization or demonstration',
    color: 'orange',
    icon: '📦',
    common_variations: ['Hero shot', 'In-use', 'Detail', 'Comparison', 'Unboxing'],
  },
  offer: {
    label: 'Offer',
    description: 'Promotional offer, pricing, or incentive',
    color: 'red',
    icon: '🏷️',
    common_variations: ['Discount', 'Bundle', 'Free trial', 'Limited time', 'Exclusive'],
  },
  cta: {
    label: 'CTA',
    description: 'Call to action driving next step',
    color: 'pink',
    icon: '👆',
    common_variations: ['Shop now', 'Learn more', 'Sign up', 'Get started', 'Book now'],
  },
  background: {
    label: 'Background',
    description: 'Visual backdrop or environment',
    color: 'slate',
    icon: '🖼️',
    common_variations: ['Solid', 'Gradient', 'Lifestyle', 'Abstract', 'Brand pattern'],
  },
  logo: {
    label: 'Logo',
    description: 'Brand logo treatment',
    color: 'slate',
    icon: '®',
    common_variations: ['Primary', 'Reversed', 'Stacked', 'Horizontal', 'Animated'],
  },
  legal: {
    label: 'Legal',
    description: 'Disclaimers and compliance text',
    color: 'slate',
    icon: '⚖️',
    common_variations: ['Standard', 'Industry-specific', 'Promotional', 'Financial'],
  },
  audio: {
    label: 'Audio',
    description: 'Voiceover, music, or sound effects',
    color: 'cyan',
    icon: '🔊',
    common_variations: ['Voiceover', 'Music bed', 'SFX', 'Sonic logo'],
  },
  end_card: {
    label: 'End Card',
    description: 'Closing frame with brand lockup',
    color: 'indigo',
    icon: '🎬',
    typical_duration: '2-3s',
    common_variations: ['Standard', 'With offer', 'With CTA', 'Animated'],
  },
  transition: {
    label: 'Transition',
    description: 'Motion between content modules',
    color: 'violet',
    icon: '↔️',
    typical_duration: '0.5-1s',
    common_variations: ['Cut', 'Fade', 'Wipe', 'Zoom', 'Slide'],
  },
};

// ============================================================================
// MODULE DEFINITION
// ============================================================================

export type ModuleFormat = 'text' | 'image' | 'video' | 'audio' | 'html5' | 'lottie';
export type SourceType = 'new_shoot' | 'existing_asset' | 'ugc' | 'stock' | 'ai_generated' | 'template';

export interface ModuleVariation {
  id: string;
  name: string;
  description?: string;
  
  // Targeting (optional - what triggers this variation)
  audience_id?: string;
  funnel_stage?: 'awareness' | 'consideration' | 'conversion' | 'retention';
  trigger?: string;
  
  // Content preview
  content_preview?: string;        // Text or description
  asset_url?: string;              // Preview image/video URL
  
  // Status
  status: 'planned' | 'in_production' | 'in_review' | 'approved' | 'live';
}

export interface Module {
  id: string;
  type: ModuleType;
  name: string;
  description: string;
  
  // Variations
  variations: ModuleVariation[];
  
  // Technical specifications
  format: ModuleFormat;
  specs: {
    dimensions?: string;           // e.g., "1080x1920"
    duration?: string;             // e.g., "3s"
    character_limit?: number;      // For text modules
    file_size_limit?: string;      // e.g., "2MB"
    frame_rate?: number;           // For video
    audio_specs?: string;          // e.g., "44.1kHz stereo"
  };
  
  // Source and reuse
  source_type: SourceType;
  dam_reference?: string;          // Link to DAM asset
  reuse_count: number;             // How many cells use this module
  used_in_cells: string[];         // Content matrix cell IDs
  
  // Metadata
  created_at: string;
  updated_at: string;
  owner?: string;
  notes?: string;
}

// ============================================================================
// DECISIONING LOGIC
// ============================================================================

export type ConditionType = 'audience' | 'funnel_stage' | 'trigger' | 'platform' | 'placement' | 'daypart' | 'geo' | 'weather' | 'custom';
export type Operator = 'equals' | 'not_equals' | 'contains' | 'not_contains' | 'in' | 'not_in' | 'greater_than' | 'less_than';

export interface DecisionCondition {
  type: ConditionType;
  field?: string;                  // For custom conditions
  operator: Operator;
  value: string | string[] | number;
}

export interface DecisionRule {
  id: string;
  name: string;
  description?: string;
  priority: number;                // Evaluation order (lower = higher priority)
  
  // Condition (when to apply this rule)
  conditions: DecisionCondition[];
  condition_logic: 'AND' | 'OR';   // How to combine multiple conditions
  
  // Action (what to show)
  action: {
    module_type: ModuleType;
    module_id: string;
    variation_id: string;
  };
  
  // Status
  is_active: boolean;
}

export interface DecisioningLogic {
  rules: DecisionRule[];
  
  // Default path (fallback when no rules match)
  defaults: {
    [key in ModuleType]?: {
      module_id: string;
      variation_id: string;
    };
  };
  
  // Validation
  has_orphan_rules: boolean;
  coverage_percentage: number;     // % of matrix cells covered by rules
}

// ============================================================================
// PLATFORM COMPATIBILITY
// ============================================================================

export type PlatformId = 
  | 'flashtalking'
  | 'innovid'
  | 'clinch'
  | 'celtra'
  | 'storyteq'
  | 'google_studio'
  | 'sizmek'
  | 'jivox'
  | 'adform'
  | 'mediamath_creative';

export interface PlatformCapabilities {
  id: PlatformId;
  name: string;
  
  // Supported formats
  supports_display: boolean;
  supports_video: boolean;
  supports_interactive: boolean;
  supports_ctv: boolean;
  supports_audio: boolean;
  supports_social: boolean;
  
  // Capabilities
  real_time_decisioning: boolean;
  sequential_messaging: boolean;
  ab_testing: boolean;
  feed_based_versioning: boolean;
  dynamic_text: boolean;
  dynamic_images: boolean;
  dynamic_video: boolean;
  
  // Constraints
  max_creative_weight?: string;    // e.g., "200KB"
  max_video_length?: string;       // e.g., "60s"
  max_feed_rows?: number;
  max_variations_per_asset?: number;
  
  // Integration
  has_api: boolean;
  feed_format: 'csv' | 'json' | 'xml' | 'api_only';
}

export const PLATFORM_LIBRARY: PlatformCapabilities[] = [
  {
    id: 'flashtalking',
    name: 'Flashtalking',
    supports_display: true,
    supports_video: true,
    supports_interactive: true,
    supports_ctv: true,
    supports_audio: false,
    supports_social: true,
    real_time_decisioning: true,
    sequential_messaging: true,
    ab_testing: true,
    feed_based_versioning: true,
    dynamic_text: true,
    dynamic_images: true,
    dynamic_video: true,
    max_creative_weight: '200KB',
    max_video_length: '120s',
    has_api: true,
    feed_format: 'csv',
  },
  {
    id: 'innovid',
    name: 'Innovid',
    supports_display: true,
    supports_video: true,
    supports_interactive: true,
    supports_ctv: true,
    supports_audio: true,
    supports_social: false,
    real_time_decisioning: true,
    sequential_messaging: true,
    ab_testing: true,
    feed_based_versioning: true,
    dynamic_text: true,
    dynamic_images: true,
    dynamic_video: true,
    max_video_length: '60s',
    has_api: true,
    feed_format: 'json',
  },
  {
    id: 'clinch',
    name: 'Clinch',
    supports_display: true,
    supports_video: true,
    supports_interactive: true,
    supports_ctv: true,
    supports_audio: false,
    supports_social: true,
    real_time_decisioning: true,
    sequential_messaging: true,
    ab_testing: true,
    feed_based_versioning: true,
    dynamic_text: true,
    dynamic_images: true,
    dynamic_video: true,
    has_api: true,
    feed_format: 'csv',
  },
  {
    id: 'celtra',
    name: 'Celtra',
    supports_display: true,
    supports_video: true,
    supports_interactive: true,
    supports_ctv: true,
    supports_audio: false,
    supports_social: true,
    real_time_decisioning: true,
    sequential_messaging: true,
    ab_testing: true,
    feed_based_versioning: true,
    dynamic_text: true,
    dynamic_images: true,
    dynamic_video: true,
    max_creative_weight: '300KB',
    has_api: true,
    feed_format: 'json',
  },
  {
    id: 'storyteq',
    name: 'Storyteq',
    supports_display: true,
    supports_video: true,
    supports_interactive: false,
    supports_ctv: true,
    supports_audio: true,
    supports_social: true,
    real_time_decisioning: false,
    sequential_messaging: true,
    ab_testing: true,
    feed_based_versioning: true,
    dynamic_text: true,
    dynamic_images: true,
    dynamic_video: true,
    has_api: true,
    feed_format: 'csv',
  },
];

// ============================================================================
// PRODUCTION TICKET
// ============================================================================

export type TicketPriority = 'critical' | 'high' | 'medium' | 'low';
export type TicketStatus = 'backlog' | 'ready' | 'in_progress' | 'in_review' | 'approved' | 'delivered';

export interface ProductionTicket {
  id: string;
  ticket_number: string;           // e.g., "PROD-2026-001"
  
  // What to build
  module_id: string;
  module_type: ModuleType;
  module_name: string;
  variations_to_create: string[];  // Variation IDs
  
  // Context
  campaign_id: string;
  campaign_name: string;
  audience_context: string[];
  placement_context: string[];
  
  // Output specifications
  output_specs: {
    format: string;
    dimensions: string;
    file_type: string;
    color_space?: string;
    max_file_size?: string;
  }[];
  
  // Priority and timeline
  priority: TicketPriority;
  due_date?: string;
  estimated_hours?: number;
  
  // Workflow
  status: TicketStatus;
  assignee?: string;
  reviewer?: string;
  
  // Downstream destination
  destination_platform?: PlatformId;
  feed_column_mapping?: {
    column_name: string;
    maps_to: 'variation_id' | 'asset_url' | 'text_content' | 'custom';
    custom_source?: string;
  }[];
  
  // Metadata
  created_at: string;
  updated_at: string;
  created_by?: string;
  notes?: string;
}

// ============================================================================
// FEED STRUCTURE
// ============================================================================

export interface FeedColumn {
  id: string;
  name: string;
  display_name: string;
  type: 'text' | 'url' | 'number' | 'date' | 'boolean';
  is_required: boolean;
  is_dynamic: boolean;              // Does this vary per row?
  source: 'module' | 'audience' | 'placement' | 'rule' | 'static' | 'formula';
  source_reference?: string;        // Module ID, audience field, etc.
  default_value?: string;
  validation_pattern?: string;
}

export interface FeedStructure {
  id: string;
  name: string;
  target_platform: PlatformId;
  
  columns: FeedColumn[];
  
  // Row generation rules
  row_per: 'audience' | 'placement' | 'audience_x_placement' | 'cell';
  include_defaults: boolean;
  
  // Validation
  estimated_row_count: number;
  validation_errors: string[];
}

// ============================================================================
// MODULE LIBRARY STATE
// ============================================================================

export interface ModuleLibraryState {
  modules: Module[];
  decisioning: DecisioningLogic;
  target_platforms: PlatformId[];
  feed_structures: FeedStructure[];
  production_tickets: ProductionTicket[];
}

File: backend/app/schemas/modules.py
"""
Module Library Schemas
For ModCon production automation and DCO platform handoff
"""
from __future__ import annotations

from typing import List, Optional, Dict, Any, Literal
from enum import Enum
from pydantic import BaseModel, Field


# ============================================================================
# MODULE TAXONOMY
# ============================================================================

class ModuleType(str, Enum):
    HOOK = "hook"
    VALUE_PROP = "value_prop"
    PROOF_POINT = "proof_point"
    PRODUCT = "product"
    OFFER = "offer"
    CTA = "cta"
    BACKGROUND = "background"
    LOGO = "logo"
    LEGAL = "legal"
    AUDIO = "audio"
    END_CARD = "end_card"
    TRANSITION = "transition"


class ModuleFormat(str, Enum):
    TEXT = "text"
    IMAGE = "image"
    VIDEO = "video"
    AUDIO = "audio"
    HTML5 = "html5"
    LOTTIE = "lottie"


class SourceType(str, Enum):
    NEW_SHOOT = "new_shoot"
    EXISTING_ASSET = "existing_asset"
    UGC = "ugc"
    STOCK = "stock"
    AI_GENERATED = "ai_generated"
    TEMPLATE = "template"


class FunnelStage(str, Enum):
    AWARENESS = "awareness"
    CONSIDERATION = "consideration"
    CONVERSION = "conversion"
    RETENTION = "retention"


class VariationStatus(str, Enum):
    PLANNED = "planned"
    IN_PRODUCTION = "in_production"
    IN_REVIEW = "in_review"
    APPROVED = "approved"
    LIVE = "live"


# ============================================================================
# MODULE VARIATION
# ============================================================================

class ModuleVariation(BaseModel):
    """A single variation of a module, potentially targeted to specific audiences."""
    
    id: str
    name: str
    description: Optional[str] = None
    
    # Targeting
    audience_ids: List[str] = Field(default_factory=list)
    funnel_stage: Optional[FunnelStage] = None
    trigger: Optional[str] = None
    
    # Content
    content_preview: Optional[str] = None
    asset_url: Optional[str] = None
    text_content: Optional[str] = None
    
    # Status
    status: VariationStatus = VariationStatus.PLANNED
    
    # Production linkage
    production_job_id: Optional[str] = None
    feed_row_ids: List[str] = Field(default_factory=list)


# ============================================================================
# MODULE SPECS
# ============================================================================

class ModuleSpecs(BaseModel):
    """Technical specifications for a module."""
    
    dimensions: Optional[str] = None
    duration: Optional[str] = None
    character_limit: Optional[int] = None
    file_size_limit: Optional[str] = None
    frame_rate: Optional[int] = None
    audio_specs: Optional[str] = None


# ============================================================================
# MODULE
# ============================================================================

class Module(BaseModel):
    """A reusable creative module in the library."""
    
    id: str
    type: ModuleType
    name: str
    description: str
    
    # Variations
    variations: List[ModuleVariation] = Field(default_factory=list)
    
    # Technical
    format: ModuleFormat
    specs: ModuleSpecs = Field(default_factory=ModuleSpecs)
    
    # Source and reuse
    source_type: SourceType = SourceType.NEW_SHOOT
    dam_reference: Optional[str] = None
    reuse_count: int = 0
    used_in_cells: List[str] = Field(default_factory=list)
    
    # Metadata
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    owner: Optional[str] = None
    notes: Optional[str] = None


# ============================================================================
# DECISIONING LOGIC
# ============================================================================

class ConditionType(str, Enum):
    AUDIENCE = "audience"
    FUNNEL_STAGE = "funnel_stage"
    TRIGGER = "trigger"
    PLATFORM = "platform"
    PLACEMENT = "placement"
    DAYPART = "daypart"
    GEO = "geo"
    WEATHER = "weather"
    CUSTOM = "custom"


class Operator(str, Enum):
    EQUALS = "equals"
    NOT_EQUALS = "not_equals"
    CONTAINS = "contains"
    NOT_CONTAINS = "not_contains"
    IN = "in"
    NOT_IN = "not_in"
    GREATER_THAN = "greater_than"
    LESS_THAN = "less_than"


class DecisionCondition(BaseModel):
    """A single condition in a decision rule."""
    
    type: ConditionType
    field: Optional[str] = None
    operator: Operator
    value: Any  # string, list, or number


class RuleAction(BaseModel):
    """What to show when a rule matches."""
    
    module_type: ModuleType
    module_id: str
    variation_id: str


class DecisionRule(BaseModel):
    """An IF/THEN rule for DCO decisioning."""
    
    id: str
    name: str
    description: Optional[str] = None
    priority: int = 100
    
    # Condition
    conditions: List[DecisionCondition] = Field(default_factory=list)
    condition_logic: Literal["AND", "OR"] = "AND"
    
    # Action
    action: RuleAction
    
    # Status
    is_active: bool = True


class DecisioningLogic(BaseModel):
    """Complete decisioning logic for a campaign."""
    
    rules: List[DecisionRule] = Field(default_factory=list)
    defaults: Dict[str, Dict[str, str]] = Field(default_factory=dict)
    has_orphan_rules: bool = False
    coverage_percentage: float = 0.0


# ============================================================================
# PLATFORM EXPORT
# ============================================================================

class PlatformId(str, Enum):
    FLASHTALKING = "flashtalking"
    INNOVID = "innovid"
    CLINCH = "clinch"
    CELTRA = "celtra"
    STORYTEQ = "storyteq"
    GOOGLE_STUDIO = "google_studio"
    SIZMEK = "sizmek"
    JIVOX = "jivox"
    ADFORM = "adform"


class FeedFormat(str, Enum):
    CSV = "csv"
    JSON = "json"
    XML = "xml"
    API_ONLY = "api_only"


class PlatformCapabilities(BaseModel):
    """Capabilities of a DCO/production automation platform."""
    
    id: PlatformId
    name: str
    
    # Supported formats
    supports_display: bool = False
    supports_video: bool = False
    supports_interactive: bool = False
    supports_ctv: bool = False
    supports_audio: bool = False
    supports_social: bool = False
    
    # Capabilities
    real_time_decisioning: bool = False
    sequential_messaging: bool = False
    ab_testing: bool = False
    feed_based_versioning: bool = False
    dynamic_text: bool = False
    dynamic_images: bool = False
    dynamic_video: bool = False
    
    # Constraints
    max_creative_weight: Optional[str] = None
    max_video_length: Optional[str] = None
    max_feed_rows: Optional[int] = None
    max_variations_per_asset: Optional[int] = None
    
    # Integration
    has_api: bool = False
    feed_format: FeedFormat = FeedFormat.CSV


# ============================================================================
# PRODUCTION TICKET (Enhanced)
# ============================================================================

class TicketPriority(str, Enum):
    CRITICAL = "critical"
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"


class TicketStatus(str, Enum):
    BACKLOG = "backlog"
    READY = "ready"
    IN_PROGRESS = "in_progress"
    IN_REVIEW = "in_review"
    APPROVED = "approved"
    DELIVERED = "delivered"


class OutputSpec(BaseModel):
    """Output specification for a production ticket."""
    
    format: str
    dimensions: str
    file_type: str
    color_space: Optional[str] = None
    max_file_size: Optional[str] = None


class FeedColumnMapping(BaseModel):
    """Maps production output to feed columns."""
    
    column_name: str
    maps_to: Literal["variation_id", "asset_url", "text_content", "custom"]
    custom_source: Optional[str] = None


class ProductionTicket(BaseModel):
    """Enhanced production ticket with full traceability."""
    
    id: str
    ticket_number: str
    
    # What to build
    module_id: str
    module_type: ModuleType
    module_name: str
    variations_to_create: List[str] = Field(default_factory=list)
    
    # Context
    campaign_id: Optional[str] = None
    campaign_name: Optional[str] = None
    audience_context: List[str] = Field(default_factory=list)
    placement_context: List[str] = Field(default_factory=list)
    
    # Output specifications
    output_specs: List[OutputSpec] = Field(default_factory=list)
    
    # Priority and timeline
    priority: TicketPriority = TicketPriority.MEDIUM
    due_date: Optional[str] = None
    estimated_hours: Optional[float] = None
    
    # Workflow
    status: TicketStatus = TicketStatus.BACKLOG
    assignee: Optional[str] = None
    reviewer: Optional[str] = None
    
    # Downstream destination
    destination_platform: Optional[PlatformId] = None
    feed_column_mapping: List[FeedColumnMapping] = Field(default_factory=list)
    
    # Traceability
    feed_row_ids: List[str] = Field(default_factory=list)
    concept_ids: List[str] = Field(default_factory=list)
    
    # Metadata
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    created_by: Optional[str] = None
    notes: Optional[str] = None


# ============================================================================
# FEED STRUCTURE
# ============================================================================

class FeedColumnType(str, Enum):
    TEXT = "text"
    URL = "url"
    NUMBER = "number"
    DATE = "date"
    BOOLEAN = "boolean"


class FeedColumnSource(str, Enum):
    MODULE = "module"
    AUDIENCE = "audience"
    PLACEMENT = "placement"
    RULE = "rule"
    STATIC = "static"
    FORMULA = "formula"


class FeedColumn(BaseModel):
    """Definition of a column in the feed export."""
    
    id: str
    name: str
    display_name: str
    type: FeedColumnType = FeedColumnType.TEXT
    is_required: bool = False
    is_dynamic: bool = True
    source: FeedColumnSource = FeedColumnSource.MODULE
    source_reference: Optional[str] = None
    default_value: Optional[str] = None
    validation_pattern: Optional[str] = None


class FeedStructure(BaseModel):
    """Structure definition for platform-specific feed export."""
    
    id: str
    name: str
    target_platform: PlatformId
    
    columns: List[FeedColumn] = Field(default_factory=list)
    
    # Row generation
    row_per: Literal["audience", "placement", "audience_x_placement", "cell"] = "audience_x_placement"
    include_defaults: bool = True
    
    # Validation
    estimated_row_count: int = 0
    validation_errors: List[str] = Field(default_factory=list)


# ============================================================================
# MODULE LIBRARY STATE
# ============================================================================

class ModuleLibraryState(BaseModel):
    """Complete state of the module library for a campaign."""
    
    modules: List[Module] = Field(default_factory=list)
    decisioning: DecisioningLogic = Field(default_factory=DecisioningLogic)
    target_platforms: List[PlatformId] = Field(default_factory=list)
    feed_structures: List[FeedStructure] = Field(default_factory=list)
    production_tickets: List[ProductionTicket] = Field(default_factory=list)


# ============================================================================
# PLATFORM EXPORT TEMPLATES
# ============================================================================

class ExportRow(BaseModel):
    """Generic export row that can be formatted per platform."""
    
    row_id: str
    creative_id: str
    creative_name: str
    
    # Module content
    modules: Dict[str, Dict[str, Any]] = Field(default_factory=dict)
    
    # Targeting
    audience_id: Optional[str] = None
    audience_name: Optional[str] = None
    funnel_stage: Optional[FunnelStage] = None
    placement_id: Optional[str] = None
    placement_name: Optional[str] = None
    geo_targeting: Optional[str] = None
    
    # Decisioning
    rules_applied: List[str] = Field(default_factory=list)
    is_default: bool = False
    
    # Technical
    dimensions: Optional[str] = None
    format: Optional[str] = None
    
    # Tracking
    destination_url: Optional[str] = None
    utm_params: Optional[str] = None
    
    # Traceability
    production_job_id: Optional[str] = None
    concept_id: Optional[str] = None


class PlatformExport(BaseModel):
    """Complete export package for a DCO platform."""
    
    platform: PlatformId
    export_format: FeedFormat
    exported_at: str
    
    # Content
    rows: List[ExportRow] = Field(default_factory=list)
    decisioning_rules: List[DecisionRule] = Field(default_factory=list)
    
    # Validation
    total_rows: int = 0
    validation_passed: bool = False
    validation_errors: List[str] = Field(default_factory=list)
    warnings: List[str] = Field(default_factory=list)
    
    # Metadata
    campaign_name: Optional[str] = None
    version: str = "1.0"
